### ISSUE TITLE
Can't run interpreter/compiler with paths with accents

### ISSUE THREAD

**Issue by bsaleil** [gambit/gambit#212]
```
$ ls /home/bapt/Téléchargements/copy-permanent.scm -l
-rw-r--r-- 1 bapt users 6199 22 sept. 14:14 /home/bapt/Téléchargements/copy-permanent.scm

$ gsc /home/bapt/Téléchargements/copy-permanent.scm 
*** ERROR IN compile-file -- No such file or directory
(path-normalize "/home/bapt/T\303\251l\303\251chargements/copy-permanent.scm")

$ gsi /home/bapt/Téléchargements/copy-permanent.scm 
*** ERROR IN ##main -- No such file or directory
(load "/home/bapt/T\303\251l\303\251chargements/copy-permanent.scm")
```

**Comment by feeley** [comment]
Please evaluate this and give the output: ```(directory-files "/home/bapt")``` (the important point is the filename string for Téléchargements)

**Comment by bsaleil** [comment]
```
> (directory-files "/home/bapt")
("Bureau" "T\351l\351chargements" "Public" "Vid\351os" "Images" "Musique" "Documents" "Mod\350les")
```

**Comment by feeley** [comment]
and ```gsi -e "(pp (command-line))(exit)" Téléchargements```

**Comment by bsaleil** [comment]
```
("gsi" "-e" "(pp (command-line))(exit)" "T\303\251l\303\251chargements")
```

**Comment by feeley** [comment]
Please try to build Gambit again with ```./configure --enable-path-encoding=latin1``` (the default is to use utf8, but apparently your filesystem uses LATIN1 encoding). But this is only one issue, because the command-line appears to use UTF8 encoding, but Gambit thinks it is LATIN1.  To fix this problem you need to change line 393 of lib/os_base.c so that it calls the function ```___NONNULLUTF_8STRINGLIST_to_SCMOBJ```. Let me know if this fixes the problem.  If so I will modify the configure script so that it automatically detects the correct encodings.

**Comment by feeley** [comment]
Forget my first suggestion... the path encoding seems correct (the result of ```directory-files```), so you don't need the ```--enable-path-encoding=latin1```. However, the command-line encoding needs to be changed.

**Comment by feeley** [comment]
Can you give some details on your environment... which OS, which terminal emulator, which shell. I suspect the terminal emulator and/or shell can be setup to encode command line arguments in LATIN1 or UTF-8. If that is true, then how can the Gambit process know which terminal emulator and shell invoked the process, and how they encode the command line arguments.

**Comment by bsaleil** [comment]
I'm using Linux, with the _gnome-terminal_ terminal emulator and _bash_.
That's weird because gnome-terminal is configured to use UTF-8, and my system uses UTF-8 too:
```
$ echo $LANG
fr_FR.UTF-8
```
Then command line arguments are not supposed to be encoded in LATIN1.
I can try you suggestion to change line 393 of lib/os_base.c, but I have no idea how to use the ___NONNULLUTF_8STRINGLIST_to_SCMOBJ function.

**Comment by feeley** [comment]
Your test shows that they are encoded in UTF-8, which prompted my suggestion to use ___NONNULLUTF_8STRINGLIST_to_SCMOBJ in lib/os_base.c.  If you change $LANG to a LATIN1 encoding, does it change what Gambit sees?

**Comment by feeley** [comment]
Fixed by commit 21c0da30d56d39f01df3df128eec33b3902bf3a3
